### PR TITLE
Add back support for non-concurrent_safe_batch saving

### PR DESCRIPTION
**Pull request by agrare**
This adds back support for the save modes other than concurrent_safe batch (required by ManageIQ) and deletion other than mark&sweep (also required by ManageIQ).

Master (now v0.3.z) was always intended to maintain compatibility with ManageIQ but along the way it diverged too far in what was required by the application's database (unique constraints, foreign-key cascade delete triggers, etc...) to be able to be used by MIQ.

As MIQ moves closer to the reference architecture that topological-inventory set we can pull these back from the v0.3.z branch where possible.

This also allows us to add support for newer versions of active_record going forward, 6.0 support already was added to v0.2.z and we're looking at supporting 6.1 and 7.0 in the future

### CODE REVIEW

**Comment by agrare** [comment]
cc @jrafanie @kbrock for rails 6.1 upgrade

File: inventory_refresh.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "activerecord", "~> 5.0"
+  spec.add_dependency "activerecord", ">=5.0", "< 6.1"
```
Comment (jrafanie) [inline inventory_refresh.gemspec:25]: Will this be loosened to drop the upper bound or "< 7.0" in a future PR?

File: lib/inventory_refresh/application_record_iterator.rb [hunk 2]
```diff
@@ -17,19 +25,28 @@ def initialize(inventory_collection: nil)
...
+          yield(batch)
+        end
+      elsif query
+        manager_uuids_set.each_slice(batch_size) do |batch|
+          yield(query.where(inventory_collection.targeted_selection_for(batch)))
+        end
+      else
+        manager_uuids_set.each_slice(batch_size) do |batch|
+          yield(inventory_collection.db_collection_for_comparison_for(batch))
+        end
```
Comment (jrafanie) [inline lib/inventory_refresh/application_record_iterator.rb:41]: Are these batch saving changes from a specific commit or series of commits that we can revert instead?  It's hard to tell if this a revert or new code.

File: inventory_refresh.gemspec (hunk 1 above)
Comment (agrare) [reply inventory_refresh.gemspec:25]: Yes

File: lib/inventory_refresh/application_record_iterator.rb (hunk 2 above)
Comment (agrare) [reply lib/inventory_refresh/application_record_iterator.rb:41]: Not new code, I can try reverting every PR between 0.3.z and 0.2.z instead

Comment (agrare) [reply lib/inventory_refresh/application_record_iterator.rb:41]: Okay done, there were ~20 PRs that had to be reverted and I had to cherry-pick NickL's rails 6 support PR which went straight to 0.2.z for some reason

Comment (jrafanie) [reply lib/inventory_refresh/application_record_iterator.rb:41]: I'm open to whatever path is easier to track and do.  It was more a question as it looked like reverts of feature commits.

Comment (agrare) [reply lib/inventory_refresh/application_record_iterator.rb:41]: Yeah reverting each is definitely easier to see what happened

**Comment by Fryguy** [comment]
@agrare Will this then be cut as a 1.0 release?

**Comment by agrare** [comment]
@Fryguy yes I figure once we get the rest of the rails 6.1 changes in I'd cut a v1.0.0

**Comment by agrare** [comment]
Think a `rubocop -A` might be in the cards as well :laughing:

**Comment by agrare** [comment]
Cross repo tests with all MIQ providers: https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/580

**Comment by Fryguy** [comment]
spec/schema.rb is like a huge chunk of those ...can probably ignore that file since it's autogenerated

**Comment by agrare** [comment]
Okay cross-repo is green

**Pull request merged**